lib/pure/math.nim

```diff
@@ -847,7 +847,16 @@ else: # JS
   func floor*(x: float64): float64 {.importc: "Math.floor", nodecl.}
   func ceil*(x: float32): float32 {.importc: "Math.ceil", nodecl.}
   func ceil*(x: float64): float64 {.importc: "Math.ceil", nodecl.}
-  func round*(x: float): float {.importc: "Math.round", nodecl.}
+
+  when not declared(copySign) or defined(nimLegacyJsRound):
+    func round*(x: float): float {.importc: "Math.round", nodecl.}
+  else:
+    func round*[T: float64 | float32](x: T): T =
+      let magnitude = abs(x)
```

Comment:
simpler:
```nim
        result = roundImpl(x)
        if x < 0 and result - x >= 0.5: result = result - 1
```
or, (micro-optimization in the case where roundImpl would be slower than ceil in js):
```nim
        if x >= 0: result = roundImpl(x)
        else:
          result = ceil(x)
          if result - x >= 0.5: result = result - 1
```

with:
```nim
        func roundImpl(x: float): float {.importc: "Math.round", nodecl.}
```

Reply:
thanks
```nim
  when (NimMajor, NimMinor) < (1, 5) or defined(nimLegacyJsRound):
    func round*(x: float): float {.importc: "Math.round", nodecl.}
  else:
    func jsRound(x: float): float {.importc: "Math.round", nodecl.}
    func round*[T: float64 | float32](x: T): T =
      if x >= 0: result = jsRound(x)
      else:
        result = ceil(x)
        if result - x >= T(0.5):
          result -= T(1.0)
```